Fill.py

```diff
@@ -196,6 +196,11 @@ def fill_restrictive(multiworld: MultiWorld, base_state: CollectionState, locati
 
     if not allow_partial and len(unplaced_items) > 0 and len(locations) > 0:
         # There are leftover unplaceable items and locations that won't accept them
+        players = set([item.player for item in unplaced_items]) | set([loc.player for loc in locations])
```

Comment:
```suggestion
        players = set(item.player for item in unplaced_items) | set(loc.player for loc in locations)
```

How about we don't create lists just to discard them instantly?